src/MongoDB.Driver.Core/Core/Authentication/ScramCache.cs

```diff
@@ -0,0 +1,128 @@
+/* Copyright 2018–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+
+namespace MongoDB.Driver.Core.Authentication
+{
+    /// <summary>
+    /// A cache for Client and Server keys, to be used during authentication.
+    /// </summary>
+    internal class ScramCache
+    {
+        private Object _cacheKey;
+        private Object _cacheValue;
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public Object Get(Object key)
```

Comment:
Suggestions:

1. Return a value of type CacheValue instead of object so caller doesn't have to downcast
2. Use TryGet pattern and return bool instead of Get and returning null

Reply:
Done (I used the TryGet pattern)